deployment/hasura/migrations/Aerie/10_goal_invocations/down.sql

```diff
@@ -0,0 +1,93 @@
+-- restore goal_analysis_* tables
+
+-- drop new PKs for created / satisfying activities
+-- and add old columns back so we can populate and re-PK
+alter table scheduler.scheduling_goal_analysis_satisfying_activities
+  drop constraint satisfying_activities_primary_key,
+  drop constraint satisfying_activities_references_scheduling_goal_analysis,
+
+  -- temp set as nullable so we can insert, made not null by PK constraint below
+  add column goal_id integer null,
+  add column goal_revision integer null;
+
+alter table scheduler.scheduling_goal_analysis_created_activities
+  drop constraint created_activities_primary_key,
+  drop constraint created_activities_references_scheduling_goal_analysis,
+
+  add column goal_id integer null,
+  add column goal_revision integer null;
+
+update scheduler.scheduling_goal_analysis_satisfying_activities as sa
+set goal_id = ga.goal_id, goal_revision = ga.goal_revision
+from scheduler.scheduling_goal_analysis ga
+where (sa.analysis_id, sa.goal_invocation_id) = (ga.analysis_id, ga.goal_invocation_id);
+
+update scheduler.scheduling_goal_analysis_created_activities as ca
+set goal_id = ga.goal_id, goal_revision = ga.goal_revision
+from scheduler.scheduling_goal_analysis ga
+where (ca.analysis_id, ca.goal_invocation_id) = (ga.analysis_id, ga.goal_invocation_id);
+
+alter table scheduler.scheduling_goal_analysis_satisfying_activities
+  add constraint satisfying_activities_primary_key
+    primary key (analysis_id, goal_id, goal_revision, activity_id),
+
+  add constraint satisfying_activities_references_scheduling_goal
+    foreign key (goal_id, goal_revision)
+      references scheduler.scheduling_goal_definition
+      on update cascade
+      on delete cascade,
+
+  drop column goal_invocation_id;
+
+alter table scheduler.scheduling_goal_analysis_created_activities
+  add constraint created_activities_primary_key
+    primary key (analysis_id, goal_id, goal_revision, activity_id),
+
+  add constraint created_activities_references_scheduling_goal
+    foreign key (goal_id, goal_revision)
+      references scheduler.scheduling_goal_definition
+      on update cascade
+      on delete cascade,
+
+  drop column goal_invocation_id;
+
+alter table scheduler.scheduling_goal_analysis
+  drop constraint scheduling_goal_analysis_primary_key;
+
+-- delete all but the first goal invocation from analysis table if there nonuniqueness
+-- between the composite key that will become the new (old) PK
+delete
+from scheduler.scheduling_goal_analysis
+where goal_invocation_id not in (select min(goal_invocation_id)
```

Comment:
We tend to run into errors when using `IN` instead of `= ANY` (or in this case `!= ANY`). If you have successfully run this migration against a DB with data that would've deleted "duplicate" invocations, however, this comment can be resolved.

Reply:
I ran the original combined migration against an `aerie-dev` dump and a local db with data and didn't see any issues. I'll rerun those checks once these migrations are finalized to make sure nothing changed.

What errors did we see? Just not matching when it should?

Reply:
Type errors that caused the migration to fail.